src/Workspaces/CSharp/Portable/LanguageServices/CSharpSyntaxFactsService.cs

```diff
@@ -559,6 +559,10 @@ public void GetNameAndArityOfSimpleName(SyntaxNode node, out string name, out in
             }
         }
 
+        public bool LooksGeneric(SyntaxNode simpleName)
+            => simpleName.IsKind(SyntaxKind.GenericName) ||
+               simpleName.GetLastToken().GetNextToken().Kind() == SyntaxKind.LessThanToken;
```

Comment:
I suppose we could.  But it seems so unlikely that you'd have code like:

```c#
foo
<...
```

As ```<``` just really isn't ever the start of anything in the language.  If, for example, we were checking if the next token was something like an identifier, i'd def be more on board about validating it was on the same line.  

What do you think?